dropwizard-jersey/src/test/java/io/dropwizard/jersey/jackson/JsonProcessingExceptionMapperTest.java

```diff
@@ -40,10 +38,10 @@ protected void configureClient(ClientConfig config) {
     }
 
     @Test
-    public void returnsA500ForNonDeserializableRepresentationClasses() throws Exception {
+    public void returnsA400ForNonDeserializableRepresentationClasses() throws Exception {
         Response response = target("/json/broken").request(MediaType.APPLICATION_JSON)
                 .post(Entity.entity(new BrokenRepresentation(ImmutableList.of("whee")), MediaType.APPLICATION_JSON));
-        assertThat(response.getStatus()).isEqualTo(500);
+        isA400ForNonDeserializableRequest(response);
```

Comment:
Nope, not in this case. It should be a 500 error because the client meets the input spec, except the server is unable to deserialize it because of the [BrokenRepresentation](https://github.com/dropwizard/dropwizard/blob/bf528890def6be0394f038be98282c541dab689f/dropwizard-jersey/src/test/java/io/dropwizard/jersey/jackson/BrokenRepresentation.java) (ie, the BrokenRepresentation can be fixed with a `@JsonCreator`, for instance)